tuf/ngclient/updater.py

```diff
@@ -471,7 +503,67 @@ def _preorder_depth_first_walk(
         # If this point is reached then target is not found, return None
         return None
 
+    def _download_directory(self, folder: str) -> None:
+        # Downloading all metadata inside the specified folder
+
+        path = (
+            f"{BASE_URL}/repository_data/TAP 14"  # File path to download from
+        )
+        path = os.path.join(path, folder)  # Add the folder to the path
+
+        # Repository might be storing data in different formats.
+
+        files = next(os.walk(path))[2]  # Get all the files inside the list
+        for file in files:
+            if file.endswith(".json"):
+                self._download_metadata(file)
+
 
 def _ensure_trailing_slash(url: str) -> str:
     """Return url guaranteed to end in a slash"""
     return url if url.endswith("/") else f"{url}/"
+
+
+def _get_spec_version(
+    repository_versions: List[str],
+    spec_version: str,
+    supported_versions: List[str],
+) -> str:
+    """Returns the specification version to be used."""
+
+    repository_versions = [int(i) for i in repository_versions]
```

Comment:
This may throw an exception; please note in the docstring

```
Raises:
   ValueError: supported_versions, repository_version or spec_version contains an invalid entry (not parseable as ``int()``)
```

Reply:
I'm also guessing that I'm not supposed to keep this in a `try-catch` block then and keep it as it is.

Reply:
Yes, you can keep it as-is. No need to re-raise. Just document that sometimes `ValueError` happens.